ext/adventure-helper/src/main/java/eu/cloudnetservice/ext/adventure/AdventureTextFormatLookup.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-2022 CloudNetService team & contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.cloudnetservice.ext.adventure;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.NonNull;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import net.kyori.adventure.text.format.TextFormat;
+import org.jetbrains.annotations.Nullable;
+
+public final class AdventureTextFormatLookup {
+
+  private static final String ALL_CHARS;
+  private static final List<TextFormat> FORMATS;
+
+  static {
+    Map<TextFormat, String> formats = new LinkedHashMap<>(16 + 5); // colors + decorations
+    // colors
+    formats.put(NamedTextColor.BLACK, "0");
+    formats.put(NamedTextColor.DARK_BLUE, "1");
+    formats.put(NamedTextColor.DARK_GREEN, "2");
+    formats.put(NamedTextColor.DARK_AQUA, "3");
+    formats.put(NamedTextColor.DARK_RED, "4");
+    formats.put(NamedTextColor.DARK_PURPLE, "5");
+    formats.put(NamedTextColor.GOLD, "6");
+    formats.put(NamedTextColor.GRAY, "7");
+    formats.put(NamedTextColor.DARK_GRAY, "8");
+    formats.put(NamedTextColor.BLUE, "9");
+    formats.put(NamedTextColor.GREEN, "a");
+    formats.put(NamedTextColor.AQUA, "b");
+    formats.put(NamedTextColor.RED, "c");
+    formats.put(NamedTextColor.LIGHT_PURPLE, "d");
+    formats.put(NamedTextColor.YELLOW, "e");
+    formats.put(NamedTextColor.WHITE, "f");
+    // decorations
+    formats.put(TextDecoration.OBFUSCATED, "k");
+    formats.put(TextDecoration.BOLD, "l");
+    formats.put(TextDecoration.STRIKETHROUGH, "m");
+    formats.put(TextDecoration.UNDERLINED, "n");
+    formats.put(TextDecoration.ITALIC, "o");
+
+    // copy into immutable, this method prevents us from (un-) boxing in runtime
+    FORMATS = List.copyOf(formats.keySet());
```

Comment:
We should really re-check if this stays always in order or if we should use something else here